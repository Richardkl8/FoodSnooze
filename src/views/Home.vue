<template>
    <div class="home">
        <RecipeCard></RecipeCard>
    </div>
</template>

<script>
    import RecipeCard from '@/components/RecipeCard.vue';

    export default {
        name: 'Home',
        components: {
            RecipeCard,
        },
    };
</script>
