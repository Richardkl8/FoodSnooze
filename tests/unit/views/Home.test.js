import { shallowMount } from '@vue/test-utils';
import Home from '../../../src/views/Home.vue';
import stubs from '../stubs/stubs';

const responseRandomRecipes = [
    {
        idMeal: '12',
        strMealThumb: 'image.jpg',
        strMeal: 'tuna',
        strCategory: 'fish',
    },
    {
        idMeal: '12',
        strMealThumb: 'image.jpg',
        strMeal: 'tuna',
        strCategory: 'fish',
    },
    {
        idMeal: '12',
        strMealThumb: 'image.jpg',
        strMeal: 'tuna',
        strCategory: 'fish',
    },
];

const provide = {
    apiService: {
        getRandomRecipes: jest.fn().mockResolvedValue(responseRandomRecipes),
    },
};


describe('Home page', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallowMount(Home, {
            provide,
            stubs: stubs.general,
        });
    });

    test('Should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Should call getRandomRecipes when component is created', async () => {
        const spy = jest.spyOn(Home.methods, 'getRandomRecipes');
        shallowMount(Home, {
            provide,
            stubs: stubs.general,
        });
        expect(spy).toHaveBeenCalled();
    });

    test('Should populate the recipes array when getRandomRecipes is called', async () => {
        expect(wrapper.vm.$data.recipes).toBe(responseRandomRecipes);
    });

    test('Should call getRandomRecipes when snooze button is clicked', async () => {
        const spy = jest.spyOn(wrapper.vm, 'getRandomRecipes');
        const snoozeButton = wrapper.find('[testId="snooze-button-home"]');
        snoozeButton.trigger('click');
        expect(spy).toHaveBeenCalledTimes(1);
    });
});
