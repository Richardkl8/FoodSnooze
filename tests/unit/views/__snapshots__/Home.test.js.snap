// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Should match snapshot 1`] = `
<div class="container">
  <div class="row my-5">
    <div class="col">
      <custom-h1-stub><b>Welcome to FoodSnooze.</b></custom-h1-stub>
      <custom-h3-stub>
        Don't know what to cook? Hit <b>snooze</b>, and let us choose!
      </custom-h3-stub>
    </div>
  </div>
  <div class="row">
    <div class="col-xs-12 col-md-4">
      <router-link-stub to="/recipe/12">
        <recipe-card-stub recipe="[object Object]" isrelatedfoods="true" testid="recipe-card-home"></recipe-card-stub>
      </router-link-stub>
    </div>
    <div class="col-xs-12 col-md-4">
      <router-link-stub to="/recipe/12">
        <recipe-card-stub recipe="[object Object]" isrelatedfoods="true" testid="recipe-card-home"></recipe-card-stub>
      </router-link-stub>
    </div>
    <div class="col-xs-12 col-md-4">
      <router-link-stub to="/recipe/12">
        <recipe-card-stub recipe="[object Object]" isrelatedfoods="true" testid="recipe-card-home"></recipe-card-stub>
      </router-link-stub>
    </div>
  </div>
  <section id="snooze-button">
    <custom-button-stub testid="snooze-button-home" text="snooze" class="center"></custom-button-stub>
  </section>
</div>
`;

exports[`Home page Should match snapshot 1`] = `
<div class="container">
  <div class="row my-5">
    <div class="col">
      <custom-h1-stub><b>Welcome to FoodSnooze.</b></custom-h1-stub>
      <custom-h3-stub>
        Don't know what to cook? Hit <b>snooze</b>, and let us choose!
      </custom-h3-stub>
    </div>
  </div>
  <div class="row">
    <div class="col-xs-12 col-md-4">
      <router-link-stub to="/recipe/12">
        <recipe-card-stub recipe="[object Object]" isrelatedfoods="true" testid="recipe-card-home"></recipe-card-stub>
      </router-link-stub>
    </div>
    <div class="col-xs-12 col-md-4">
      <router-link-stub to="/recipe/12">
        <recipe-card-stub recipe="[object Object]" isrelatedfoods="true" testid="recipe-card-home"></recipe-card-stub>
      </router-link-stub>
    </div>
    <div class="col-xs-12 col-md-4">
      <router-link-stub to="/recipe/12">
        <recipe-card-stub recipe="[object Object]" isrelatedfoods="true" testid="recipe-card-home"></recipe-card-stub>
      </router-link-stub>
    </div>
  </div>
  <section id="snooze-button">
    <custom-button-stub testid="snooze-button-home" text="snooze" class="center"></custom-button-stub>
  </section>
</div>
`;
